### PR TITLE
chore: Update spacing tokens

### PR DESCRIPTION
**Pull request by alisonjoseph**
Updates to spacing tokens to use new tokens
Adds tokens across site to margin/padding

### Testing
- double check I didn't replace anything with the wrong tokens 😅  

_might be easier to review if you hide whitespace changes_

### CODE REVIEW

File: src/components/ComponentExample/component-example.scss [hunk 1]
```diff
@@ -43,7 +43,7 @@
   bottom: 0;
   color: $text-02;
   @include carbon--type-style('caption-01');
-  padding: 0.5rem 1rem;
+  padding: $spacing-03 1rem;
```
Comment (jnm2377) [inline src/components/ComponentExample/component-example.scss]: did you want to update the `1rem` to also use a spacing token?

File: src/components/PageTable/page-table.scss [hunk 2]
```diff
@@ -26,7 +26,7 @@
       td {
         @include carbon--type-style('body-short-01');
         color: $text-01;
-        padding: $spacing-md;
+        padding: $spacing-05;
 
         img[src$='checkmark--glyph.svg'] {
           width: 16px;
```
Comment (jnm2377) [inline src/components/PageTable/page-table.scss]: semi related to this PR. Would we want to also update layout tokens? If they're being used or not used, like here?

File: src/components/ComponentExample/component-example.scss (hunk 1 above)
Comment (alisonjoseph) [reply src/components/ComponentExample/component-example.scss]: good catch 👍

File: src/components/PageTable/page-table.scss (hunk 2 above)
Comment (alisonjoseph) [reply src/components/PageTable/page-table.scss]: We're not currently using layout tokens anywhere on the site I don't think, but would probably make sense to do that at some point.

**Pull request merged**